### PR TITLE
chore(CPlusPlus): add exponential search

### PR DESCRIPTION
**Pull request by Ashborn-SM**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
Add exponential search

**Other information:**

### CODE REVIEW

File: algorithms/CPlusPlus/README.md [hunk 1]
```diff
@@ -33,6 +33,8 @@
 4. [Finding squareroot using Binary search](Searching/sqrt-monotonic-binary-search.cpp)
 5. [KMP String Searching](Searching/kmp.cpp)
 6. [Ternary Search](Searching/Ternary-search.cpp)
+7. [Interpolation Search](Searching/interpolation-search.cpp)
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/README.md:36]: You have added 2 files here but I can only view the Exponential Search file. The interpolation Search file is missing.

**Review by aayushjain7 — CHANGES_REQUESTED** [review]

File: algorithms/CPlusPlus/README.md (hunk 1 above)
Comment (Ashborn-SM) [reply algorithms/CPlusPlus/README.md:36]: The interpolation search is already there in the repo, the contributor must have forgot to add it the readme

Comment (aayushjain7) [reply algorithms/CPlusPlus/README.md:36]: Okayy! That's great!

**Review by aayushjain7 — APPROVED** [review]
Good work! 💯

**Comment by ming-tsai** [comment]
Nice work

**Pull request merged**